### ISSUE TITLE
NullPointerException with default servlet, include and welcome pages

### ISSUE THREAD

**Issue by grgrzybek** [eclipse/jetty.project#9966]
**Jetty version(s)**
12.0.0 SNAPSHOT

**Description**
**How to reproduce?**

I added a comment under #9927, but here's separate issue.

@gregw I can't track the exact part of 0b1c28a8883bb70421a228d1a72644d6125f2baa, but I think it started causing NPE in some special case.

My servlet is invoked with `/gateway/x?what=include&where=/`:
```java
String what = req.getParameter("what");
String where = req.getParameter("where");
switch (what) {
	case "redirect":
		resp.sendRedirect(where);
		return;
	case "forward":
		// we can't send anything when forwarding
		req.getRequestDispatcher(where).forward(req, resp);
		return;
	case "include":
		resp.getWriter().print(">>>");
		req.getRequestDispatcher(where).include(req, resp);
		resp.getWriter().print("<<<");
		return;
	default:
}
```

`org.eclipse.jetty.ee10.servlet.ServletApiResponse#getWriter()` calls:
```java
String encoding = getServletResponseInfo().getCharacterEncoding(true);
```

`org.eclipse.jetty.ee10.servlet.ServletContextResponse#getCharacterEncoding(boolean)` calls:
```java
if (setContentType)
    setCharacterEncoding(encoding, EncodingFrom.DEFAULT);
```

so `org.eclipse.jetty.ee10.servlet.ServletContextResponse#_characterEncoding` is set to `iso-8859-1`.
Because the forward is for `/`, default servlet is involved (because I need to test welcome files).

`org.eclipse.jetty.ee10.servlet.DefaultServlet#doGet()` has this fragment:
```java
if (contextResponse != null)
{
    String characterEncoding = contextResponse.getRawCharacterEncoding();
    if (characterEncoding != null)
        content = new ForcedCharacterEncodingHttpContent(content, characterEncoding);
}
```

and the constructor is:
```java
public ForcedCharacterEncodingHttpContent(HttpContent content, String characterEncoding)
{
    super(content);
    this.characterEncoding = characterEncoding;
    String mimeType = content.getContentTypeValue();
    int idx = mimeType.indexOf(";charset");
    if (idx >= 0)
        mimeType = mimeType.substring(0, idx);
    this.contentType = mimeType + ";charset=" + this.characterEncoding;
}
```

the `content` is:
```
content = {org.ops4j.pax.web.service.jetty.internal.web.DefaultServlet$UnknownLengthHttpContent@4857} "UnknownLengthHttpContent@c898d8f[CompressedFormatsHttpContent@1c61db46[ResourceHttpContent@471a9af8{r=file:/data/sources/github.com/ops4j/org.ops4j.pax.web-jakarta-new/pax-web-jetty/target/b1/,ct=null}]]"
 _delegate: org.eclipse.jetty.http.content.HttpContent  = {org.eclipse.jetty.http.content.PreCompressedHttpContentFactory$CompressedFormatsHttpContent@4919} "CompressedFormatsHttpContent@1c61db46[ResourceHttpContent@471a9af8{r=file:/data/sources/github.com/ops4j/org.ops4j.pax.web-jakarta-new/pax-web-jetty/target/b1/,ct=null}]"
  _delegate: org.eclipse.jetty.http.content.HttpContent  = {org.eclipse.jetty.http.content.ResourceHttpContent@4922} "ResourceHttpContent@471a9af8{r=file:/data/sources/github.com/ops4j/org.ops4j.pax.web-jakarta-new/pax-web-jetty/target/b1/,ct=null}"
   _contentType: java.lang.String  = null
   _etag: org.eclipse.jetty.http.HttpField  = {org.eclipse.jetty.http.PreEncodedHttpField@4925} "ETag: W/"///+d0tLiXs/////73HTsQ""
   _path: java.nio.file.Path  = {sun.nio.fs.UnixPath@4924} "/data/sources/github.com/ops4j/org.ops4j.pax.web-jakarta-new/pax-web-jetty/target/b1"
   _resource: org.eclipse.jetty.util.resource.Resource  = {org.eclipse.jetty.util.resource.PathResource@4749} "file:/data/sources/github.com/ops4j/org.ops4j.pax.web-jakarta-new/pax-web-jetty/target/b1/"
  compressedFormats: java.util.Set  = {java.util.HashSet@4921}  size = 0
```

the underlying resource is directory, so the extension is `*` and the resulting content type is null.

so I'm getting NullPointerException.

My test you can run (without bothering with OSGi stuff) is https://github.com/ops4j/org.ops4j.pax.web/blob/1b26a2f8ea394d696a6780ff0fa92b2613019dff/pax-web-jetty/src/test/java/org/ops4j/pax/web/service/jetty/internal/UnifiedJettyTest.java#L361

**Comment by grgrzybek** [comment]
I think that the default servlet should do:
```java
if (contextResponse != null)
{
    String characterEncoding = contextResponse.getRawCharacterEncoding();
    if (characterEncoding != null)
        content = new ForcedCharacterEncodingHttpContent(content, characterEncoding);
}
```

only after ensuring that the content is not a directory (or do some other verification).

**Comment by lorban** [comment]
@grgrzybek I've pushed a fix (see #9970) to the jetty-12.0.x branch. If you've got any feedback, I'd be happy to hear it. Thanks!

**Comment by grgrzybek** [comment]
Works perfectly - thanks @lorban !
